lib/mayaUsd/render/mayaToHydra/renderOverride.cpp

```diff
@@ -406,21 +406,6 @@ MStatus MtohRenderOverride::Render(const MHWRender::MDrawContext& drawContext) {
         it->PreFrame(drawContext);
     }
 
-    // TODO: Is there a way to improve this? Quite silly.
-    auto enableShadows = true;
-    auto* lightParam = drawContext.getLightParameterInformation(
-        0, MHWRender::MDrawContext::kFilteredIgnoreLightLimit);
-    if (lightParam != nullptr) {
-        MIntArray intVals;
-        if (lightParam->getParameter(
-                MHWRender::MLightParameterInformation::kGlobalShadowOn,
```

Comment:
What makes this HdSt specific?

Reply:
The light Lighting and Shadow tasks are only ever created for Storm, and without those tasks those parameters do nothing.

Reply:
Looking at the [comment in Hydra](https://github.com/PixarAnimationStudios/USD/blob/ebac0a8b6703f4fa1c27115f1f013bb9819662f4/pxr/imaging/hdx/taskController.cpp#L269) and wondering how we can make sure the code in here will stay in sync with Hydra.

One option would be to add TF_CODING_ERROR in `else` case (line 478) which would validate that lighting and shadow tasks were not created?

Another option would be to keep setting these parameters outside of `_isUsingHdSt` but add a condition to check if a specific task (lighting or shadow) is created.

Either one would work for me...and would save us debugging time when Hydra changes.

Reply:
To me it seems that would just be duplicating the **if-else** outside of _taskController.cpp_ with not a clear picture of how/when it changes what would be needed (i.e. I'm not convinced those will ever be common to all delegates).

Reply:
We are making assumptions over here and matching the hardcoded choice from the task controller. It is a good habit to provide validation for these assumptions so if they ever change, we won't have to spend hours investigating what's going on. Especially that we will forget about this code as soon as it's merged.

BUT, maybe the reason you are not convinced is that there is no easy way to add this validation? I looked quickly and can't find any way to check if _shadowTaskId or _simpleLightTaskId is empty. Iterating over all tasks and casting doesn't look ideal. If that's the case, let's document it better in the source code, point to actual tasks that these parameters are affecting and what in task controller is responsible for creating them.